### ISSUE TITLE
I have a question about inWriteTransaction…

### ISSUE THREAD

**Issue by aimobier** [realm/realm-cocoa#4876]
Is it necessary for me to judge ` isinWriteTransaction` every time WriteTransaction ？

**Comment by istx25** [comment]
Hi @aimobier. I don't really follow your question. Would you be able to elaborate further about what you mean? Are you asking if you need to check `isinWriteTransaction` before committing another write transaction or something else. Thanks

**Comment by aimobier** [comment]
````
[realm beginWriteTransaction];
// Something that might take some time
[realm commitWriteTransaction];
````

Is not as long as I can guarantee matching `beginWriteTransaction` and `commitWriteTransaction` ,'The Realm is already in a write transaction' is not going to happen?
Or do I still need to judge 'isinwritetransaction' at the beginning of 'beginwritetransaction' or 'commitwritetransaction'?

chinese: 我可以保证每一次开始写入都会对应一个提交写入，在这种情况下。是不是还会有机会存在已经正在写入的崩溃呢？  我需要在每次写入之前都需要进行 是不是 正在写 的判断呢？

**Comment by istx25** [comment]
I would recommend still checking `isInWriteTransaction` before trying to begin another write transaction. Though, there is a grey area. If you think beyond a reasonable doubt that there will only ever be one write transaction happening/trying to happen at once, then I don't think it's necessary but in the name of safety, it's probably still a good idea to have it.

Does that answer your question? Let me know if you have any follow-up questions.

**Comment by jpsim** [comment]
No, please don't check `isInWriteTransaction` for every write transaction you want to perform.

Write transactions are designed to be a logical grouping of operations that should only be performed atomically together. This way, if your app crashes half-way through a write transaction, you don't end up with inconsistent state at your model layer.

In other words, you should keep write transactions localized whenever possible.

There are situations in which you want longer-running write transactions, which is usually when you'd use `beginWrite()` and `commitWrite()`.

If you're writing a function that needs to mutate Realm objects both from inside and outside write transactions, it's in that case that you'd need to verify `isInWriteTransaction` prior to calling `beginWrite()` and `commitWrite()`.

So to recap, conditionally starting/committing write transactions should be the exception, not the norm.

**Comment by istx25** [comment]
I stand corrected. JP knows more about this. I've crossed out my previous comment.

**Comment by aimobier** [comment]
Thank you very much for the perfect solution to my problem.
